src/github.com/matrix-org/dendrite/appservice/consumers/roomserver.go

```diff
@@ -165,13 +173,20 @@ func (s *OutputRoomEventConsumer) lookupStateEvents(
 // each namespace of each registered application service, and if there is a
 // match, adds the event to the queue for events to be sent to a particular
 // application service.
-func (s *OutputRoomEventConsumer) filterRoomserverEvents(ctx context.Context, events []gomatrixserverlib.Event) error {
+func (s *OutputRoomEventConsumer) filterRoomserverEvents(
+	ctx context.Context,
+	events []gomatrixserverlib.Event,
+) error {
 	for _, event := range events {
 		for _, appservice := range appServices {
 			// Check if this event is interesting to this application service
 			if s.appserviceIsInterestedInEvent(ctx, event, appservice) {
-				// TODO: Queue this event to be sent off to the application service
-				fmt.Println(appservice.ID, "was interested in", event.Sender(), event.Type(), event.RoomID())
+				// Queue this event to be sent off to the application service
+				if err := s.asDB.StoreEvent(ctx, appservice.ID, event); err != nil {
+					log.WithError(err).Warn("failed to insert incoming event into appservices database")
+				} else {
+					ecm[appservice.ID]++
```

Comment:
concurrent map writes, yikes!

Reply:
This'll thankfully no longer be necessary with the new event-based method of sending events Erik proposed.